pmoves/services/hi-rag-gateway-v2/clients/qdrant.py

```
@@ -0,0 +1,170 @@
+    raise AttributeError("qdrant client has no compatible search/query API")
+
+
+def ensure_qdrant_collection(vector_dim: int):
+    """Create or resync the Qdrant collection when the embed dimension changes.
+
+    When QDRANT_RECREATE_ON_DIM_MISMATCH=false (default), a dimension mismatch
+    returns HTTP 409 instead of silently destroying indexed vectors.
+    """
+    try:
+        info = qdrant.get_collection(COLL)
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)